### PR TITLE
Use Rollup

### PR DESCRIPTION
**Pull request by gkatsev**
Based partially on https://github.com/videojs/video.js/compare/master...Rich-Harris:rollup?expand=1
Currently, exports UMD, minified UMD, CJS, and ES rolluped files as well as equivalent formats for the novtt file.
Still needs to be updated to work with the tests, which could be an issue, potentially, because of proxyquireify. Maybe tests will stay as they are.

The build is currently failing because the only babelrc is told to not compile module syntax and that is causing grunt and things to fail.

### CODE REVIEW

**Comment by gkatsev** [comment]
Here's the assets output:
```sh
┌───────────────────────────────┬───────────┬───────────┐
│ filename                      │      size │   gzipped │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video-js-cdn.css          │  53.12 KB │  13.99 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video-js-cdn.min.css      │   45.6 KB │  13.69 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.js            │ 796.66 KB │ 151.63 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.min.js        │ 226.18 KB │  47.63 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.rollup.cjs.js │  604.6 KB │ 134.85 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.rollup.es.js  │ 604.58 KB │ 134.84 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.rollup.js     │ 604.83 KB │ 134.93 KB │
├───────────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.rollup.min.js │ 156.56 KB │  40.42 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video-js.css                  │  53.31 KB │     14 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video-js.min.css              │  45.74 KB │  13.69 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video-js.swf                  │  16.72 KB │  16.74 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.js                      │ 853.01 KB │ 165.83 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.min.js                  │ 245.82 KB │  53.97 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.rollup.cjs.js           │  732.1 KB │ 154.19 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.rollup.es.js            │ 732.08 KB │ 154.18 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.rollup.js               │ 732.33 KB │ 154.26 KB │
├───────────────────────────────┼───────────┼───────────┤
│ video.rollup.min.js           │ 213.41 KB │  49.48 KB │
└───────────────────────────────┴───────────┴───────────┘
```

all the files that rollup currently generates have rollup in the name and generally correspond to the same files that don't have rollup.

One thing I just realized is that the cjs builds shouldn't be running with node-resolve and commonjs plugins so bundlers will pick those up. The ES one should potentially do that as well.

**Comment by gkatsev** [comment]
Will require this change in vttjs: https://github.com/videojs/vtt.js/pull/14

File: build/rollup.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  dest: 'dist/video.js'
+};
+
+const minifiedUmd = Object.assign({}, _.cloneDeep(umd), {
+  dest: 'dist/video.min.js'
+});
+
+minifiedUmd.options.plugins.splice(4, 0, uglify({
+  preserveComments: 'some',
+  srewIE8: false,
```
Comment (misteroneill) [inline build/rollup.js]: Appears to be a typo.

Comment (gkatsev) [inline build/rollup.js]: Yup.

File: package.json [hunk 2]
```diff
@@ -128,13 +150,14 @@
     "videojs-doc-generator": "0.0.1",
     "videojs-flash": "^1.1.0",
     "videojs-standard": "^6.0.1",
-    "webpack": "^2.3.0"
+    "webpack": "^1.15.0"
```
Comment (gkatsev) [inline package.json]: webpack 2 doesn't support IE8: https://github.com/webpack/webpack/issues/3070

**Comment by gkatsev** [comment]
I believe this is good to go now. One way to test and verify this is to clone this branch, npm install and run `npm run build`. Then, clone [videojs-bundler-sample](https://github.com/videojs/videojs-bundler-sample) and in it run `npm install /path/to/videojs` followed by `npm start`. Then, if you go to the index page in the bundler repo in the browser, you'll be able to see it with all the various types of bundlers. Doing `npm install` with a path over linking is necessary because otherwise babel ends up breaking for some of the builds.

**Comment by gkatsev** [comment]
The current output of the assets script:
```
┌──────────────────────────┬───────────┬───────────┐
│ filename                 │      size │   gzipped │
├──────────────────────────┼───────────┼───────────┤
│ alt/video-js-cdn.css     │  53.12 KB │  13.98 KB │
├──────────────────────────┼───────────┼───────────┤
│ alt/video-js-cdn.min.css │   45.6 KB │  13.69 KB │
├──────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.js       │ 611.79 KB │ 136.34 KB │
├──────────────────────────┼───────────┼───────────┤
│ alt/video.novtt.min.js   │ 158.16 KB │  40.79 KB │
├──────────────────────────┼───────────┼───────────┤
│ video-js.css             │  53.31 KB │     14 KB │
├──────────────────────────┼───────────┼───────────┤
│ video-js.min.css         │  45.74 KB │  13.69 KB │
├──────────────────────────┼───────────┼───────────┤
│ video-js.swf             │  16.91 KB │  16.93 KB │
├──────────────────────────┼───────────┼───────────┤
│ video.cjs.js             │ 600.68 KB │ 133.11 KB │
├──────────────────────────┼───────────┼───────────┤
│ video.es.js              │ 600.43 KB │ 133.05 KB │
├──────────────────────────┼───────────┼───────────┤
│ video.js                 │ 664.96 KB │ 150.79 KB │
├──────────────────────────┼───────────┼───────────┤
│ video.min.js             │ 177.58 KB │  47.35 KB │
└──────────────────────────┴───────────┴───────────┘
```

**Comment by gkatsev** [comment]
Here's a merged table:
```
┌──────────────────────────┬─────────────────┬─────────────┬────────────────────┬────────────────┐
│ filename                 │ size browserify │ size rollup │ gzipped browserify │ gzipped rollup │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ alt/video-js-cdn.css     │        53.12 KB │    53.12 KB │           13.99 KB │       13.98 KB │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ alt/video-js-cdn.min.css │         45.6 KB │     45.6 KB │           13.69 KB │       13.69 KB │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ alt/video.novtt.js       │       796.66 KB │   611.79 KB │          151.63 KB │      136.34 KB │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ alt/video.novtt.min.js   │       226.18 KB │   158.16 KB │           47.63 KB │       40.79 KB │
├──────────────────────────┼─────────────────┤─────────────┼────────────────────┼────────────────┤
│ video-js.css             │        53.31 KB │    53.31 KB │              14 KB │          14 KB │
├──────────────────────────┼─────────────────┤─────────────┼────────────────────┼────────────────┤
│ video-js.min.css         │        45.74 KB │    45.74 KB │           13.69 KB │       13.69 KB │
├──────────────────────────┼─────────────────┤─────────────┼────────────────────┼────────────────┤
│ video-js.swf             │        16.72 KB │    16.91 KB │           16.74 KB │       16.93 KB │
├──────────────────────────┼─────────────────┤─────────────┼────────────────────┼────────────────┤
│ video.cjs.js             │ N/A             │   600.68 KB │ N/A                │      133.11 KB │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ video.es.js              │ N/A             │   600.43 KB │ N/A                │      133.05 KB │
├──────────────────────────┼─────────────────┼─────────────┼────────────────────┼────────────────┤
│ video.js                 │       853.01 KB │   664.96 KB │          165.83 KB │      150.79 KB │
├──────────────────────────┼─────────────────┤─────────────┼────────────────────┼────────────────┤
│ video.min.js             │       245.82 KB │   177.58 KB │           53.97 KB │       47.35 KB │
└──────────────────────────┴─────────────────┴─────────────┴────────────────────┴────────────────┘
```

File: build/rollup.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      args.progress ? progress() : {},
+      filesize()
+    ],
+    onwarn(warning) {
+      if (warning.code === 'UNUSED_EXTERNAL_IMPORT') {
+        return;
+      }
+      if (warning.code === 'UNRESOLVED_IMPORT') {
+        return;
+      }
```
Comment (misteroneill) [inline build/rollup.js]: Maybe these two blocks should be combined.

**Pull request merged**